bob/src/cluster/quorum.rs

```diff
@@ -351,22 +437,78 @@ impl Cluster for Quorum {
+                trace!("EXIST {} keys check local node", len);
+                let local_keys = local.collect(keys);
+                match exist_on_local_node(&self.backend, &local_keys).await {
+                    Ok(local_exist) => {
+                        trace!("EXIST {} keys check local node: found {}/{} keys",
+                               len, local_exist.len(), local.len());
+                        local.update_existence(&mut result, &local_exist);
+                    },
+                    Err(e) => warn!("EXIST {} check local node failed: {:?}", len, e)
+                };
+            }
+        }
+
+        trace!("EXIST {} keys check primary nodes", len);
+        match Self::collect_remote_exists(&mut result, keys, &mut primary).await {
+            Ok(_) => trace!("EXIST {} keys check primary nodes finished", len),
+            Err(e) => warn!("EXIST {} keys check primary nodes failed: {:?}", len, e)
+        }
+
+        trace!("EXIST {} keys check secondary nodes", len);
+        match Self::collect_remote_exists(&mut result, keys, &mut secondary).await {
+            Ok(_) => trace!("EXIST {} keys check secondary nodes finished", len),
+            Err(e) => warn!("EXIST {} keys check secondary nodes failed: {:?}", len, e)
+        }
+
+        let mut alien_index_map = IndexMap::where_not_exists(&result);
+
+        if !alien_index_map.is_empty() {
+            trace!("EXIST {} keys check local alien", len);
+            match exist_on_local_alien(&self.backend, &alien_index_map.collect(keys)).await {
+                Ok(local_alien_result) => {
+                    trace!("EXIST {} keys check local alien finished: found {}/{} keys",
+                           len, local_alien_result.len(), alien_index_map.len()); 
+                    alien_index_map.update_existence(&mut result, &local_alien_result);
+                    alien_index_map.retain_not_existed(&result)
+                },
+                Err(e) => warn!("EXIST {} keys check local alien failed: {:?}", len, e)
+            }
+        }
+
+        if !alien_index_map.is_empty() {
+            trace!("EXIST {} keys check remote alien", len);
+            let all_remote_nodes: Vec<_> = self
+                .mapper
+                .nodes()
+                .iter()
+                .filter(|n| n.name() != self.mapper.local_node_name())
+                .collect();
+            let remote_nodes_aliens_exist =
+                exist_on_remote_aliens(&all_remote_nodes, &alien_index_map.collect(keys)).await;
+            trace!("EXIST {} keys check remote alien finished", len);
+            for remote_alien_result in remote_nodes_aliens_exist {
+                match remote_alien_result {
+                    Ok(remote_alien_result) => {
+                        alien_index_map.update_existence(&mut result, remote_alien_result.inner());
+                    }
+                    Err(e) => warn!("EXIST {} keys check remote alien failed on node {}: {:?}", 
```

Comment:
Maybe it is better to log at the `debug` level, since if one node is unavailable, every request will write a log at `wran` level and overflow the log files